feat(linter/eslint-plugin-vitest): reuse no-mocks-import jest linter rule

Afsoon opened the pull request:
Related to #4656

Implement [no-mocks-import](https://github.com/vitest-dev/eslint-plugin-vitest/blob/main/docs/rules/no-mocks-import.md) lint ruler adding a new entry in the mapping between jest and vitest compatible rules.

Oxlint-migrate PR related: https://github.com/oxc-project/oxlint-migrate/pull/269

Afsoon committed:
feat(linter/eslint-plugin-vitest): reuse the jest linter rule

connorshea commented on crates/oxc_linter/src/utils/mod.rs:
```diff
@@ -33,7 +33,7 @@ pub use self::{
 /// List of Jest rules that have Vitest equivalents.
 // When adding a new rule to this list, please ensure oxlint-migrate is also updated.
 // See https://github.com/oxc-project/oxlint-migrate/blob/2c336c67d75adb09a402ae66fb3099f1dedbe516/scripts/constants.ts
```
Should make sure to open a PR to oxlint-migrate if we end up merging this one

connorshea reviewed (dismissed):
The docs block in the `declare_oxc_lint!` should be updated to mention that this rule is compatible with Vitest as well. (we should probably consider auto-generating something in the docs for these cases as well, honestly, but for now this is how it is done).

Other rules have a note like so:
```rs
    /// This rule is compatible with [eslint-plugin-vitest](https://github.com/veritem/eslint-plugin-vitest/blob/v1.1.9/docs/rules/no-alias-methods.md),
    /// to use it, add the following configuration to your `.oxlintrc.json`:
    ///
    /// ```json
    /// {
    ///   "rules": {
    ///      "vitest/no-alias-methods": "error"
    ///   }
    /// }
    /// ```
```

Afsoon committed:
feat(linter/eslint-plugin-vitest): Mention that this rule can be reuse with vitest

camc314 reviewed (changes requested):
Thanks for working on this!

Do you mind also porting the vitest test cases for this rule to ensure that we're catching all cases?

This can be done by:
```
just new-vitest-rule no-mocks-import
```

And copying over the generated test cases to `crates/oxc_linter/src/rules/jest/no_mocks_import.rs`, this allows us to ensure that we're reporting all cases correctly for both vitest/jest

e.g. https://github.com/oxc-project/oxc/blob/4911d6073810965150fd0e344101701a434ad7ec/crates/oxc_linter/src/rules/jest/consistent_test_it.rs#L794-L795

Afsoon committed:
feat(linter/eslint-plugin-vitest): Extend test with vitest test cases and message based on framework

Afsoon commented:
@camc314 Updated the PR with the vitests test case but I have notice the diagnostic message is framework dependant. I'm not sure how to approach here:
- The current approach
- One diagnostic method but add a new argument, for example:
```
let caller = {
  if ctx.frameworks().is_vitest() {
      "vi"
  }

 "jest"
};

fn no_mocks_import_diagnostic(span: Span, call: &str) -> OxcDiagnostic {
    OxcDiagnostic::warn("Mocks should not be manually imported from a `__mocks__` directory.")
        .with_help(format!("Instead use `{call}.mock` and import from the original module path."))
        .with_label(span)
}
```
- Duplicate the rule under the vitest folder.

camc314 committed:
u

framwork flags can sometime be unreliable, so merge the two help messages into one, and update the help text

camc314 reviewed (approved):
Thank yoiu!
